src/MockNet/Http/Contents/ObjectContent.cs

```diff
@@ -0,0 +1,85 @@
+using System;
+using System.Text;
+using SystemHttpContent = System.Net.Http.HttpContent;
+using SystemStringContent = System.Net.Http.StringContent;
+
+namespace Theorem.MockNet.Http
+{
+    public class ObjectContent : IHttpContent
+    {
+        internal readonly object value;
+
+        private readonly Type type;
+        private readonly Encoding encoding;
+        private readonly string mediaType;
+
+        public ObjectContent(Type type, object content)
+        {
+            this.type = type;
+            this.value = content;
+            this.encoding = Encoding.UTF8;
+            this.mediaType = "application/json";
+        }
+
+        public SystemHttpContent ToHttpContent() => new SystemStringContent(Utils.Json.ToString(value), encoding, mediaType);
+
+        #region Overrides
+        public override string ToString()
+        {
+            return value.ToString();
+        }
+
+        public override int GetHashCode()
+        {
+            return value.GetHashCode();
+        }
+
+        public override bool Equals(object obj)
+        {
+            if (obj is null)
+            {
+                return false;
+            }
+
+            if (obj.GetType() == type)
+            {
+                return this.value.Equals(obj);
+            }
+
+            {
```

Comment:
Extra brackets here and on line 56?

Reply:
Actually, I put them there so I can use the same variable `content` in two different sections.

Reply:
If we're targetting C# 8, these two scopes and the final `return` could be replaced with:

```csharp
            var other = obj switch
            {
                ObjectContent c => c.value,
                SystemStringContent c => Utils.Json.ToObject(c.ReadAsStringAsync().GetAwaiter().GetResult(), type),
                _ => throw new NotImplementedException(),
            };

            return this.value.Equals(other);
```

I imagine that it's even possible to add the `null` and `type == obj.Gettype()` comparisons to the statement too.

Reply:
Actually, the following works and passes all tests:

```csharp
           var other = obj switch
            {
                ObjectContent c => c.value,
                SystemStringContent c => Utils.Json.ToObject(c.ReadAsStringAsync().GetAwaiter().GetResult(), type),
                _ => obj,
            };

            return value.Equals(other);
```

Reply:
I like the switch statement expression here. switching.

Reply:
Updated